### PR TITLE
chore: add cache control to web console files not to cache index html

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [x] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/cloudfront-s3-portal.ts [hunk 1]
```diff
@@ -188,6 +189,9 @@ export class CloudFrontS3Portal extends Construct {
       prune: false,
       distribution: this.distribution,
       distributionPaths: props.frontendProps.autoInvalidFilePaths,
+      cacheControl: [
+        CacheControl.maxAge(Duration.days(1)),
```
Comment (zxkane) [inline src/control-plane/cloudfront-s3-portal.ts:193]: Will it set the cache control of all files to 1 day?

If so, I prefer to create another BucketDeployment instance to upload `index.html` only with 1-day cache control.

File: src/control-plane/cloudfront-s3-portal.ts [hunk 2]
```diff
@@ -168,29 +170,63 @@ export class CloudFrontS3Portal extends Construct {
...
+    } else {
+      const commands = [...props.frontendProps.buildCommands];
+      if (excludeIndexHtml) {
+        commands.push('cd /asset-output');
+        commands.push('rm /asset-output/index.html');
+      } else {
+        commands.push('cd /asset-output');
+        commands.push('ls | grep -xv index.html | xargs rm -rf');
+      }
+      return Source.asset(props.frontendProps.assetPath, {
```
Comment (zxkane) [inline src/control-plane/cloudfront-s3-portal.ts:213]: How about sharing the same source but using [exclude](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3_deployment.BucketDeploymentProps.html#exclude) and `include` `index.html`?

Creating two sources probably slows down the build of `cdk synth`.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/cloudfront-s3-portal.ts (hunk 2 above)
Comment (tyyzqmf) [reply src/control-plane/cloudfront-s3-portal.ts:213]: Has no effect on Assets bundled using the `bundling` property.
Ref: https://github.com/aws/aws-cdk/blob/7c62d6809cbdeee251c8ff31e22db7d0ff85aba2/packages/aws-cdk-lib/core/lib/fs/options.ts#L64

Comment (zxkane) [reply src/control-plane/cloudfront-s3-portal.ts:213]: Thanks for sharing it.

**Review by zxkane — APPROVED** [review]

**Pull request merged**